docs/book/beyond_basics/sample_webhook.md

```diff
@@ -0,0 +1,207 @@
+# Webhook Example
+
+This chapter walks through a simple webhook implementation.
+
+It uses the [controller-runtime](https://godoc.org/sigs.k8s.io/controller-runtime/pkg) libraries to implement
+the Webhook and Manager. The webhook can run with other controllers in the same manager.
+
+## Setup
+
+#### Way to Deploy your Webhook Server
+
+There are various ways to deploy the webhook server in terms of
+
+1. Where the serving certificates live.
+1. In what environment the webhook server runs, in a pod or directly on a VM, etc.
+1. If in a pod, on what type of node, worker nodes or master node.
+
+The recommended way to deploy the webhook server is
+
+1. Run the webhook server as a regular pod on worker nodes through a workload API, e.g. Deployment or StatefulSet.
+1. Put the certificate in a k8s secret in the same namespace as the webhook server
+1. Mount the secret as a volume in the pod
+1. Create a k8s service to front the webhook server.
+
+#### Creating a Webhook Handler
+
+{% method %}
+
+`podAnnotator` is an admission webhook handler. It implements the `admission.Handler` interface.
+Details about how to implement an admission webhook podAnnotator is covered in a later section.
+
+{% sample lang="go" %}
+```go
+type podAnnotator struct {
+	client  client.Client
+	decoder types.Decoder
+}
+
+// podAnnotator implements admission.Handler.
+var _ admission.Handler = &podAnnotator{}
+
+func (a *podAnnotator) Handle(ctx context.Context, req types.Request) types.Response {
+	...
+}
+```
+{% endmethod %}
+
+
+#### Building a Webhook with the Handler
+
+{% method %}
+
+[controller-runtime](https://godoc.org/sigs.k8s.io/controller-runtime/pkg) provides a useful package for
```

Comment:
Introduce the Webhook.  e.g.

`
A Webhook configures what type of requests the Handler should accept from the apiserver.  Options include:
- The type of the Operations (CRUD)
- The type of the Targets (Deployment, Pod, etc)
- The type of the Handler (Mutating, Validating)

When the server starts, the Webhook will register itself with the apiserver to start accepting requests.
`